In [1]:
import glob
import os
import numpy as np 

## 0) loading the l1c files

In [2]:
#path = '/home/datawork-cersat-public/cache/project/sarwave/data/products/tests/iw/slc/l1c/4.1/'
#safe = 'S1A_IW_XSP__1SDV*.SAFE'
path = '/home/datawork-cersat-public/project/sarwave/data/products/tests2/slc/iw/l1c/'
safe = '*IW_XSP__1SDV*B07*.SAFE'


z = glob.glob(path+'*/*/'+safe)
fns = glob.glob(path+'*/*/'+safe+'/*b07*.nc')  # the iw2 sub_swaths have a 5 tile_sample whereas the 2 others (iw1 et iw3) have a 4 tile_sample
print('number of SAFE files : '+str(len(z)), '; number of nc files : '+str(len(fns))) 

fns_array = np.array(fns)
# np.savetxt('/home1/datawork/ljessel/l1c_conversion/4.1/ALL_S1A_IW_XSP__1SDV.txt', fns_array, fmt='%s')
np.savetxt('/home1/datawork/ljessel/l1c_conversion/ALL_S1AB_IW_XSP__1SDV_B07.txt', fns_array, fmt='%s')
fns_array

number of SAFE files : 9007 ; number of nc files : 26940


array(['/home/datawork-cersat-public/project/sarwave/data/products/tests2/slc/iw/l1c/2018/263/S1A_IW_XSP__1SDV_20180920T122004_20180920T122032_023782_02980D_BF31_B07.SAFE/l1c-s1a-iw2-xsp-1sdv-20180920t122005-20180920t122030-023782-02980d-b07.nc',
       '/home/datawork-cersat-public/project/sarwave/data/products/tests2/slc/iw/l1c/2018/263/S1A_IW_XSP__1SDV_20180920T122004_20180920T122032_023782_02980D_BF31_B07.SAFE/l1c-s1a-iw3-xsp-1sdv-20180920t122006-20180920t122031-023782-02980d-b07.nc',
       '/home/datawork-cersat-public/project/sarwave/data/products/tests2/slc/iw/l1c/2018/263/S1A_IW_XSP__1SDV_20180920T122004_20180920T122032_023782_02980D_BF31_B07.SAFE/l1c-s1a-iw1-xsp-1sdv-20180920t122004-20180920t122032-023782-02980d-b07.nc',
       ...,
       '/home/datawork-cersat-public/project/sarwave/data/products/tests2/slc/iw/l1c/2023/308/S1A_IW_XSP__1SDV_20231104T182128_20231104T182155_051071_06288B_A9E6_B07.SAFE/l1c-s1a-iw3-xsp-1sdv-20231104t182128-20231104t182153-051071-06288b-b07.nc',


9007 SAFE et 27_042 fichiers .nc du type :  /home/datawork-cersat-public/project/sarwave/data/products/tests2/slc/iw/l1c/* IW_XSP__1SDV* B07*.SAFE/*.nc

9007 SAFE et 26_940 fichiers .nc du type :  /home/datawork-cersat-public/project/sarwave/data/products/tests2/slc/iw/l1c/* IW_XSP__1SDV* B07*.SAFE/* b07*.nc